### PR TITLE
{cosmosdb} Fix #23806: Add mandatory parameters when backup policy ty…

### LINKED ISSUE #23806: "az cosmodb create" command failed with Periodic backup policy
> ### `az cosmodb create` failed with option --backup-policy-type Periodic as of CLI version 2.40.0

**Related command**
```powershell
$backupPolicy="Periodic"
az cosmosdb create --name $dbName --resource-group $resourceGroup `
        --backup-policy-type $backupPolicy `
        --default-consistency-level Session `
        --kind MongoDB --server-version 4.2
```

**Describe the bug**
I get the error:
```
The command failed with an unexpected error. Here is the traceback:
local variable 'periodic_mode_properties' referenced before assignment
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/cosmosdb/custom.py", line 146, in cli_cosmosdb_create 
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/cosmosdb/custom.py", line 276, in _create_database_account
UnboundLocalError: local variable 'periodic_mode_properties' referenced before assignment
```
However, everything works fine with "Continuous" backup policy and if I specify `--backup-retention 24 --backup-interval 720`

**Expected behavior**
It should work even without retention and interval (with default values)? Or at least the documentation should tell they are mandatory in case of periodic backup as well as the command when only policy is specified.

**Environment summary**
Powershell script under Windows OS

### PR DESCRIPTION
**Pull request by RakeshMohanMSFT**
…pe is periodic

Fix https://github.com/Azure/azure-cli/issues/23806

**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
cosmosdb

File: src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py [hunk 1]
```diff
@@ -266,6 +266,9 @@ def _create_database_account(client,
     backup_policy = None
     if backup_policy_type is not None:
         if backup_policy_type.lower() == 'periodic':
+            if backup_interval is None and backup_retention is None:
+                raise CLIError(
+                    '--backup-interval and --backup-retention must be specified when Backup Policy Type is Periodic.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:271]: CC @amisi01 since I'm not familiar with service logic

Comment (amisi01) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:271]: if backup_interval is None and backup_retention is None and backup_redundancy is None:
        raise CLIError(
                    '--backup-interval, --backup-retention or --backup-redundancy must be specified when Backup Policy Type is Periodic.')

File: src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py [hunk 2]
```diff
@@ -266,6 +266,9 @@ def _create_database_account(client,
...
             backup_policy = PeriodicModeBackupPolicy()
             if backup_interval is not None or backup_retention is not None or backup_redundancy is not None:
                 periodic_mode_properties = PeriodicModeProperties(
```
Comment (amisi01) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:274]: Also, updatethe following code:

    elif backup_interval is not None or backup_retention is not None:
        backup_policy = PeriodicModeBackupPolicy()
        periodic_mode_properties = PeriodicModeProperties(
            backup_interval_in_minutes=backup_interval,
            backup_retention_interval_in_hours=backup_retention
        )
        backup_policy.periodic_mode_properties = periodic_mode_properties
        
To:

     elif backup_interval is not None or backup_retention is not None **_or backup_redundancy is not None_**:
          backup_policy = PeriodicModeBackupPolicy()
          periodic_mode_properties = PeriodicModeProperties(
              backup_interval_in_minutes=backup_interval,
              backup_retention_interval_in_hours=backup_retention
          )
          backup_policy.periodic_mode_properties = periodic_mode_properties

Comment (RakeshMohanMSFT) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:274]: I am not sure what the ** mean, did you mean to add commented lines?

Comment (amisi01) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:274]: that means add following condition to elif : "or backup_redundancy is not None"

Comment (RakeshMohanMSFT) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py:275]: Done

File: src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py (hunk 1 above)
Comment (RakeshMohanMSFT) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py]: Made all the suggestions

**Review by amisi01 — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**